docs/USER_GUIDE.md

```diff
@@ -339,6 +381,39 @@ We have now succesfully made different version and stored them, a user can view
 ![Version 1](images/update-step-version-1.png)
 ![Current version](images/update-step-version-cur.png)
 
+## Custom search in models
+This functionality makes it possible to find nodes based on text input of the user. It is available for Flintmodels and LawSource versions. Matched nodes are returned to the user, including their role inside the model.  
+
+Usage inside **Flintmodels**:
+- `Right-Click` on the Flintmodel and select `Search` from the dropdown menu.  
+![Flintmodel Search](images/flintmodel_search.png)  
+- Input the query inside the `text` bar.  
+![Flintmodel Search](images/flintmodel_query.png)  
+- `Click` on the results and `Go To` to open that specific node. 
+
+Usage inside **LawSources**:
+- `Right-Click` on a version below a LawSource and select `Search` from the dropdown menu.  
+![LawSource Search](images/lawsource_search.png)  
+- Input the query inside the `text` bar.  
+![LawSource Search](images/lawsource_query.png)  
+- `Click` on the results and `Go To` to open that specific node. 
+
+## Valid dates of sourceparts
+This functionality makes it possible to assign dates to each line of an Article.
+To assign dates:  
+- `Click` on the line and open the inspector  
+![Sourcepart Date](images/sourcepart_dates.png)  
+- `Click` on the calander icon of the date you want to assign and select the date.  
+![Sourcepart Select Date](images/sourcepart_select_date.png)  
+- The assigned dates will be saved and displayed in the inspector editor  
+![Sourcepart Date Result](images/sourcepart_date_result.png)  
+
+## Context Actions
+A lot of functions are accesible via intentions. These can be seen whilst pressing `Alt+Enter` on a selected node, but also via the context actions tab in the right-hand corner. The advantage of the Context Actions tab is that this tab can remain open while you are making interpretations. As soon as you select a node which has intentions the Context Actions tab will show them, removing the need of pressing `Alt+Enter` before you can see them.  
```

Comment:
From the description it sounds like the menu is only too see them. Should also mention that you can click and execute them from the menu as well.